### ISSUE TITLE
String:split regex, inconsistent with other string methods

### ISSUE THREAD

**Issue by gisikw** [KSP-KOS/KOS#1270]
I discovered that the recently-released string:split actually operates on regex strings, rather than substrings. See https://github.com/KSP-KOS/KOS/blob/e37682fb165ca184c67185663fe2f1067bf84326/src/kOS.Safe/Encapsulation/StringValue.cs#L149

Thus `"example.string":SPLIT(".")` => `LIST("","","",...)`.

That said, escape sequences are still appropriately recognized, so `"example.string":SPLIT("\.")` yields the expected `LIST("example","string")`.

Personally, I'm really excited that there's a way to access some regexp at all. It opens up the doors for some fun scripting, e.g.

```
FUNCTION ISNUMERIC {
  PARAMETER str.
  LOCAL tokens IS str:SPLIT("\d+").
  RETURN tokens:LENGTH = 2 AND tokens[0]:LENGTH = 0 AND tokens[1]:LENGTH = 0.
}
```

That said, I suspect regex is going to be non-intuitive for a non-trivial portion of the userbase, and since the other methods (such as find, replace, etc) are based on string, rather than regex arguments, it's probably best that this follow the same pattern.

(though if you wanna expose a separate regex version of the string methods, that'd be super awesome too!)

**Comment by Dunbaratu** [comment]
Hmm.  If this is fixed it should be fixed right now, else it gets out into the userbase like this and people start writing scripts expecting it this way and then we have to make the non-regex version the special case instead of the regex version.

**Comment by gisikw** [comment]
Yeah, I checked the docs, they make no mention of it accepting regular expressions. Given that all the other String methods don't accept 'em, it's probably best to remove that functionality here. Just wanted to point it out since the behavior is inconsistent between methods.

**Comment by Dunbaratu** [comment]
This creates another problem too, which is that I wanted to toy with the idea of processing backslash-strings some day so people can write things like PRINT("hello\nworld.").  But if people start expecting to use a regex printed as "\d", then it would be a backward compatibility mess later if we started supporting backslashes in such a way that "\d" has to become "\\d" to preserve the backslash long enough for it to pass into the regex parser.

**Comment by hvacengi** [comment]
I just erased 30 minutes worth of writing after remembering that we can escape the characters to treat them as literals using a Regex method:
```
        public ListValue<string> SplitToList(string separator)
        {
            string[] split = Regex.Split(internalString, Regex.Escape(separator), RegexOptions.IgnoreCase);
            return new ListValue<string>(split);
        }
```
Which should prevent the separator string from being treated as a regular expression.  I agree that a distinct regular expression method would be a good addition in the future (mentioned in the developer chat).

I'll test and verify, then submit a PR.  It should not have any affect on the docs, so I think it should be a simple fix.

**Comment by chippydip** [comment]
That seems like a lot of extra work instead of just using:

    var split = internalString.Split(new [] { separator }, StringSplitOptions.None);

**Comment by hvacengi** [comment]
@chippydip Sure, it's extra work, but it's required in order to maintain case insensitivity.  The debate on sensitivity has been hashed out enough, and I don't want to revisit it for this specific issue, I'm going to post the text that I was writing as I worked through the problem.  I cut it from the post when I remembered `Regex.Escape`, but it shows the process of how I got to this point.  I have made minor edits, but it is essentially still in draft form:

The most important thing is to make sure that single character splits will work as expected, since they are the most common thing a user is going to try and split.  And splitting on a `.` character should be an expected use case, especially since we now have a better file access system.  My first reaction is to just implement the string split method:
```
        public ListValue<string> SplitToList(string separator)
        {
            string[] split = internalString.Split(new string[] { separator }, StringSplitOptions.None );
            return new ListValue<string>(split);
        }
```
Which will perform the default expected split behavior.  But that doesn't account for the case insensitivity in the current string system.  Since we need to split on `"A"` and `"a"` at the same time, then it can be revised to:
```
        public ListValue<string> SplitToList(string separator)
        {
            string[] split = internalString.Split(new string[] { separator.ToUpper(), separator.ToLower() }, StringSplitOptions.None );
            return new ListValue<string>(split);
        }
```
Now we've introduced another issue however.  Since split accepts a string value itself, and is not limited to a single character, there is a chance that the user will pass `"AaBbCc" as the split string, which technically according to the docs would need to split for `"aaBBcc", "AabbCc", "AAbbcc", etc...`.  For that level of control, we need to use regular expressions because the built in string split method does not support case insensitivity.  This would also be one of those points where I think case sensitivity would probably be a good thing, since splitting on a string rather than a character can lead to very unexpected results.  Unless we can live with insensitivity, I think I'd figure on a combined implementation for the most flexibility:
```
        public ListValue<string> SplitToList(string separator)
        {
            if (separator.Length > 1)
            {
                string[] split = Regex.Split(internalString, separator, RegexOptions.IgnoreCase);
                return new ListValue<string>(split);
            }
            else
            {
                string[] split = internalString.Split(new string[] { separator.ToUpper(), separator.ToLower() }, StringSplitOptions.None);
                return new ListValue<string>(split);
            }
        }
```
This gives better results when calling a split on a single regex character.  What it doesn't fix the regex special characters in longer strings (like splitting on `"e.s"`).  So Ideally we need to escape the characters *[it was at this point that I remembered that Regex.Escape existed]*
